index.js

```diff
@@ -1,291 +1,24 @@
-      param.in = 'formData'
-      param.name = name
-      parameters.push(param)
-    })
-  }
-}
-
-function getPathParams (parameters, params) {
-  if (params.type && params.properties) {
-    // for the shorthand querystring declaration
-    return getPathParams(parameters, params.properties)
-  }
-
-  Object.keys(params).forEach(p => {
-    const param = {}
-    param.name = p
-    param.in = 'path'
-    param.required = true
-    param.description = params[p].description
-    param.type = params[p].type
-    parameters.push(param)
-  })
-}
-
-function getHeaderParams (parameters, headers) {
-  if (headers.type && headers.properties) {
-    // for the shorthand querystring declaration
-    const headerProperties = Object.keys(headers.properties).reduce((acc, h) => {
-      const required = (headers.required && headers.required.indexOf(h) >= 0) || false
-      const newProps = Object.assign({}, headers.properties[h], { required })
-      return Object.assign({}, acc, { [h]: newProps })
-    }, {})
-
-    return getHeaderParams(parameters, headerProperties)
-  }
-
-  Object.keys(headers).forEach(h =>
-    parameters.push({
-      name: h,
-      in: 'header',
-      required: headers[h].required,
-      description: headers[h].description,
-      type: headers[h].type
-    })
-  )
-}
-
-function genResponse (response) {
-  // if the user does not provided an out schema
-  if (!response) {
-    return { 200: { description: 'Default Response' } }
-  }
-
-  // remove previous references
-  response = Object.assign({}, response)
-
-  Object.keys(response).forEach(key => {
-    if (response[key].type) {
-      var rsp = response[key]
-      var description = response[key].description
-      var headers = response[key].headers
-      response[key] = {
-        schema: rsp
-      }
-      response[key].description = description || 'Default Response'
-      if (headers) response[key].headers = headers
-    }
-
-    if (!response[key].description) {
-      response[key].description = 'Default Response'
-    }
-  })
-
-  return response
-}
-
-// The swagger standard does not accept the url param with ':'
-// so '/user/:id' is not valid.
-// This function converts the url in a swagger compliant url string
-// => '/user/{id}'
-function formatParamUrl (url) {
-  var start = url.indexOf('/:')
-  if (start === -1) return url
-
-  var end = url.indexOf('/', ++start)
 
-  if (end === -1) {
-    return url.slice(0, start) + '{' + url.slice(++start) + '}'
-  } else {
-    return formatParamUrl(url.slice(0, start) + '{' + url.slice(++start, end) + '}' + url.slice(end))
+  // by default the mode is dynamic, as plugin initially was developed
+  opts.mode = opts.mode || 'dynamic'
+
+  switch (opts.mode) {
+    case 'static':
+      setup.static(fastify, opts, next)
+      break
+    case 'dynamic':
+      setup.dynamic(fastify, opts, next)
+      break
+    default:
+      throw new Error("unsupported mode, should be one of ['static', 'dynamic']")
```

Comment:
Can you call `next()` with this error instead?

Reply:
done, thanks